apps/emqx_resource/src/emqx_resource_health_check.erl

```diff
@@ -55,12 +58,30 @@ delete_checker(Name) ->
 	end.
 
 health_check(Name, SleepTime) ->
-    case emqx_resource:health_check(Name) of
-        ok ->
-            emqx_alarm:deactivate(Name);
-        {error, _} ->
-            emqx_alarm:activate(Name, #{name => Name},
-                <<Name/binary, " health check failed">>)
+    receive
+        {Pid, start_health_check}  ->
+            case emqx_resource:health_check(Name) of
+                ok ->
+                    emqx_alarm:deactivate(Name);
+                {error, _} ->
+                    emqx_alarm:activate(Name, #{name => Name},
+                        <<Name/binary, " health check failed">>)
+            end,
+            Pid ! health_check_finish
     end,
-    timer:sleep(SleepTime),
     health_check(Name, SleepTime).
+
+health_check_timeout_checker(Pid, Name, SleepTime) ->
+    SelfPid = self(),
+    Pid ! {SelfPid, start_health_check},
+    receive
+        health_check_finish -> timer:sleep(SleepTime)
```

Comment:
We need to monitor the health_check process, so that if it dies we can do something.